server/handlers/register_test.go

```diff
@@ -0,0 +1,92 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/lesterfernandez/roommate-finder/server/data"
+	"github.com/lesterfernandez/roommate-finder/server/token"
+)
+
+func TestRegister(t *testing.T) {
+	t.Run("New User Created", func(t *testing.T) {
+
+		mockUserRepo := data.UserRepo{
+			CreateUser: func(newUser data.RegisterBody) error { return nil },
+			UserExists: func(username string) bool { return false },
+		}
+
+		r := CreateHandler(&Server{User: &mockUserRepo})
+
+		testUser := data.RegisterBody{
+			ProfilePic:  "url",
+			DisplayName: "hank",
+			Budget:      3,
+			Gender:      "Male",
+			Cleanliness: 3,
+			Loudness:    3,
+			Coed:        true,
+			Username:    "hank",
+			Password:    "turtle",
+		}
+
+		reqBody, _ := json.Marshal(testUser)
+		req := httptest.NewRequest(http.MethodPost, "/register", bytes.NewReader(reqBody))
+		w := httptest.NewRecorder()
+		r.ServeHTTP(w, req)
+
+		res := w.Result()
+		resBody, _ := io.ReadAll(res.Body)
+		tokenRes := data.TokenMessage{}
+		unmarshalErr := json.Unmarshal(resBody, &tokenRes)
+
+		if unmarshalErr != nil {
+			t.Fatalf("Could not parse response: %v\n", unmarshalErr)
+		}
+
+		//TODO: Check for status code once we set that up
```

Comment:
I left a suggestion for this TODO. It's a small change so I recommend including it into this PR if you like.